### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by Metsanpeitto**
In this branch I have implemented these changes:
 Create a table named `vets` with the following columns:
  - `id`: integer (set it as autoincremented PRIMARY KEY)
  - `name`: string
  - `age`: integer
  - `date_of_graduation`: date
- There is a **many-to-many** relationship between the tables `species` and `vets`: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called `specializations` to handle this relationship.
- There is a **many-to-many** relationship between the tables `animals` and `vets`: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called `visits` to handle this relationship, it should also keep track of the date of the visit.
- Insert the following data for vets:
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- Insert the following data for specialties:
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.
- Insert the following data for visits:
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.
- Write queries to answer the following:
  - Who was the last animal seen by William Tatcher?
  - How many different animals did Stephanie Mendez see?
  - List all vets and their specialties, including vets with no specialties.
  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
  - What animal has the most visits to vets?
  - Who was Maisy Smith's first visit?
  - Details for most recent visit: animal information, vet information, and date of visit.
  - How many visits were with a vet that did not specialize in that animal's species?
  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -2,14 +2,70 @@
...
+    name varchar(100),
+    age INT,
+    date_of_graduation DATE NOT NULL
+);
+
+CREATE TABLE specializations (
+    vet_name varchar(100),
+    species_name varchar(100)
+);
+
```
Comment (Emmyn5600) [inline schema.sql]: - [ ] You should add a new join table with the name `specializations` with the following columns:

             ` species_id int,
               vet_id int`

File: data.sql [hunk 2]
```diff
@@ -16,3 +16,137 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
...
+/* Taks-4  */
+INSERT INTO vets (name, age, date_of_graduation ) VALUES ('William Tatcher' , 45, '2000-4-23');
+INSERT INTO vets (name, age, date_of_graduation ) VALUES ('Maisy Smith' , 26 , '2019-1-17');
+INSERT INTO vets (name, age, date_of_graduation ) VALUES ('Stephanie Mendez' , 64 , '1981-5-4');
+INSERT INTO vets (name, age, date_of_graduation ) VALUES ('Jack Harkness' , 38 , '2008-6-8');
+
+INSERT INTO specializations (vet_name, species_name) VALUES ('William Tatcher' , 'Pokemon');
+INSERT INTO specializations (vet_name, species_name) VALUES ('Stephanie Mendez' , 'Pokemon');
+INSERT INTO specializations (vet_name, species_name) VALUES ('Stephanie Mendez' , 'Digimon');
+INSERT INTO specializations (vet_name, species_name) VALUES ('Jack Harkness' , 'Digimon');
```
Comment (Emmyn5600) [inline data.sql]: - [ ] After changing the columns in the `specifications` table, kindly, Insert the following data again:

` INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'William Tatcher'), (SELECT id FROM species WHERE name = 'Pokemon')); `

 `INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Stephanie Mendez'), (SELECT id FROM species WHERE name = 'Pokemon'));`

 `INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Stephanie Mendez'), (SELECT id FROM species WHERE name = 'Digimon'));`

 `INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Jack Harkness'), (SELECT id FROM species WHERE name = 'Digimon')); `

File: queries.sql [hunk 3]
```diff
@@ -68,7 +71,99 @@ SELECT AVG(weight_kg) FROM animals;
...
+AND visits.date_of_visit > '2020-04-01' AND visits.date_of_visit < '2020-08-30' ;
+
+/*  What animal has the most visits to vets?  */
+SELECT DISTINCT animals.name, COUNT(animal_name)  FROM visits JOIN animals ON 
+visits.animal_name = animals.name
+GROUP BY animals.name ORDER BY COUNT(animal_name) DESC LIMIT 1;
+
+/*  Who was Maisy Smith's first visit?  */
+SELECT animal_name, date_of_visit FROM visits ORDER BY date_of_visit ASC LIMIT 1;
+
```
Comment (Emmyn5600) [inline queries.sql]: - [ ] This query is not displaying, who was Maisy Smith's first visit. You should have written in your solution `WHERE vets.name = 'Maisy Smith' `

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @Metsanpeitto 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:. 
   
 - The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Changes required :writing_hand: 

  - Please check the comment under the review.
  
  Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: schema.sql (hunk 1 above)
Comment (Metsanpeitto) [reply schema.sql]: Ok/ :heavy_check_mark:

File: data.sql (hunk 2 above)
Comment (Metsanpeitto) [reply data.sql]: Ok, Done! :heavy_check_mark:

File: queries.sql (hunk 3 above)
Comment (Metsanpeitto) [reply queries.sql]: Thanks a lot for your time, I really appreciate it. 
I have already solved this. :heavy_check_mark:

Comment (Metsanpeitto) [reply queries.sql]: Now It looks like this :
```cmd 
SELECT animal_name, date_of_visit FROM visits WHERE vet_name = 'Maisy Smith'  ORDER BY date_of_visit   ASC
 LIMIT 1;

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashbaord. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**